Error when fetching repository from maven zip artifact

I am using this feature introduced in tycho [2.2](https://wiki.eclipse.org/Tycho/Release_Notes/2.2#Support_for_consuming_maven_artifacts_made_of_zipped_P2_update_sites) but I am getting this exception when trying to load a repository, not sure if it is a problem with the repository or a bug in tycho.

Note that I can reproduce the issue with tycho 2.3, 2.4 and 2.5-SNAPSHOT, but I guess the issue could be in the p2 version.

```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-packaging-plugin:2.5.0-SNAPSHOT:package-feature (default-package-feature) on project com.ataccama.ide.one.feature: Execution default-package-feature of goal org.eclipse.tycho:tycho-packaging-plugin:2.5.0-SNAPSHOT:package-feature failed: Load repository from url mvn:com.ataccama.extension.salesforce:com.ataccama.extension.salesforce.ide.repository:0.2.1-SNAPSHOT:zip failed (org.eclipse.equinox.p2.core.ProvisionException: No repository found at mvn:com.ataccama.extension.salesforce:com.ataccama.extension.salesforce.ide.repository:0.2.1-SNAPSHOT:zip.) -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.tycho:tycho-packaging-plugin:2.5.0-SNAPSHOT:package-feature (default-package-feature) on project com.ataccama.ide.one.feature: Execution default-package-feature of goal org.eclipse.tycho:tycho-packaging-plugin:2.5.0-SNAPSHOT:package-feature failed: Load repository from url mvn:com.ataccama.extension.salesforce:com.ataccama.extension.salesforce.ide.repository:0.2.1-SNAPSHOT:zip failed (org.eclipse.equinox.p2.core.ProvisionException: No repository found at mvn:com.ataccama.extension.salesforce:com.ataccama.extension.salesforce.ide.repository:0.2.1-SNAPSHOT:zip.)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-package-feature of goal org.eclipse.tycho:tycho-packaging-plugin:2.5.0-SNAPSHOT:package-feature failed: Load repository from url mvn:com.ataccama.extension.salesforce:com.ataccama.extension.salesforce.ide.repository:0.2.1-SNAPSHOT:zip failed (org.eclipse.equinox.p2.core.ProvisionException: No repository found at mvn:com.ataccama.extension.salesforce:com.ataccama.extension.salesforce.ide.repository:0.2.1-SNAPSHOT:zip.)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.RuntimeException: Load repository from url mvn:com.ataccama.extension.salesforce:com.ataccama.extension.salesforce.ide.repository:0.2.1-SNAPSHOT:zip failed (org.eclipse.equinox.p2.core.ProvisionException: No repository found at mvn:com.ataccama.extension.salesforce:com.ataccama.extension.salesforce.ide.repository:0.2.1-SNAPSHOT:zip.)
    at org.eclipse.tycho.repository.p2base.artifact.repository.RepositoryArtifactProvider.loadRepository (RepositoryArtifactProvider.java:118)
    at org.eclipse.tycho.repository.p2base.artifact.repository.LazyArtifactRepository.getDelegate (LazyArtifactRepository.java:50)
    at org.eclipse.tycho.repository.p2base.artifact.repository.LazyArtifactRepository.contains (LazyArtifactRepository.java:87)
    at org.eclipse.tycho.repository.p2base.artifact.repository.RepositoryArtifactProvider.contains (RepositoryArtifactProvider.java:137)
    at org.eclipse.tycho.repository.local.MirroringArtifactProvider.contains (MirroringArtifactProvider.java:110)
    at org.eclipse.tycho.repository.p2base.artifact.provider.CompositeArtifactProvider.contains (CompositeArtifactProvider.java:59)
    at org.eclipse.tycho.repository.p2base.artifact.provider.CompositeArtifactProvider.getArtifactFile (CompositeArtifactProvider.java:86)
    at org.eclipse.tycho.p2.target.TargetPlatformBaseImpl.getArtifactLocation (TargetPlatformBaseImpl.java:142)
    at org.eclipse.tycho.p2.target.FinalTargetPlatformImpl.getArtifactLocation (FinalTargetPlatformImpl.java:1)
    at org.eclipse.tycho.packaging.FeatureXmlTransformer.expandReferences (FeatureXmlTransformer.java:65)
    at org.eclipse.tycho.packaging.PackageFeatureMojo.expandVersionQualifiers (PackageFeatureMojo.java:259)
    at org.eclipse.tycho.packaging.PackageFeatureMojo.execute (PackageFeatureMojo.java:126)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.eclipse.equinox.p2.core.ProvisionException: No repository found at mvn:com.ataccama.extension.salesforce:com.ataccama.extension.salesforce.ide.repository:0.2.1-SNAPSHOT:zip.
    at org.eclipse.equinox.internal.p2.repository.helpers.AbstractRepositoryManager.fail (AbstractRepositoryManager.java:397)
    at org.eclipse.equinox.internal.p2.repository.helpers.AbstractRepositoryManager.loadRepository (AbstractRepositoryManager.java:701)
    at org.eclipse.equinox.internal.p2.artifact.repository.ArtifactRepositoryManager.loadRepository (ArtifactRepositoryManager.java:131)
    at org.eclipse.equinox.internal.p2.artifact.repository.ArtifactRepositoryManager.loadRepository (ArtifactRepositoryManager.java:125)
    at org.eclipse.tycho.p2.remote.RemoteArtifactRepositoryManager.loadRepository (RemoteArtifactRepositoryManager.java:105)
    at org.eclipse.tycho.repository.p2base.artifact.repository.RepositoryArtifactProvider.loadRepository (RepositoryArtifactProvider.java:116)
    at org.eclipse.tycho.repository.p2base.artifact.repository.LazyArtifactRepository.getDelegate (LazyArtifactRepository.java:50)
    at org.eclipse.tycho.repository.p2base.artifact.repository.LazyArtifactRepository.contains (LazyArtifactRepository.java:87)
    at org.eclipse.tycho.repository.p2base.artifact.repository.RepositoryArtifactProvider.contains (RepositoryArtifactProvider.java:137)
    at org.eclipse.tycho.repository.local.MirroringArtifactProvider.contains (MirroringArtifactProvider.java:110)
    at org.eclipse.tycho.repository.p2base.artifact.provider.CompositeArtifactProvider.contains (CompositeArtifactProvider.java:59)
    at org.eclipse.tycho.repository.p2base.artifact.provider.CompositeArtifactProvider.getArtifactFile (CompositeArtifactProvider.java:86)
    at org.eclipse.tycho.p2.target.TargetPlatformBaseImpl.getArtifactLocation (TargetPlatformBaseImpl.java:142)
    at org.eclipse.tycho.p2.target.FinalTargetPlatformImpl.getArtifactLocation (FinalTargetPlatformImpl.java:1)
    at org.eclipse.tycho.packaging.FeatureXmlTransformer.expandReferences (FeatureXmlTransformer.java:65)
    at org.eclipse.tycho.packaging.PackageFeatureMojo.expandVersionQualifiers (PackageFeatureMojo.java:259)
    at org.eclipse.tycho.packaging.PackageFeatureMojo.execute (PackageFeatureMojo.java:126)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR]
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
```

I re-used the example from #30, it tries to build a product that references a feature in an mvn zipped update site. This comes from an Apache-licensed project that I found in central. On 2.5.0-SNAPSHOT the stack trace looks like the following:

```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-p2-repository-plugin:2.5.0-SNAPSHOT:assemble-repository (default-assemble-repository) on project issue271: Execution default-assemble-repository of goal org.eclipse.tycho:tycho-p2-repository-plugin:2.5.0-SNAPSHOT:assemble-repository failed: Load repository from url mvn:eu.scasefp7.eclipse:s-case-core-update-site:1.1.0:zip failed (org.eclipse.equinox.p2.core.ProvisionException: No repository found at mvn:eu.scasefp7.eclipse:s-case-core-update-site:1.1.0:zip.) -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.tycho:tycho-p2-repository-plugin:2.5.0-SNAPSHOT:assemble-repository (default-assemble-repository) on project issue271: Execution default-assemble-repository of goal org.eclipse.tycho:tycho-p2-repository-plugin:2.5.0-SNAPSHOT:assemble-repository failed: Load repository from url mvn:eu.scasefp7.eclipse:s-case-core-update-site:1.1.0:zip failed (org.eclipse.equinox.p2.core.ProvisionException: No repository found at mvn:eu.scasefp7.eclipse:s-case-core-update-site:1.1.0:zip.)
[...]
Caused by: java.lang.RuntimeException: Load repository from url mvn:eu.scasefp7.eclipse:s-case-core-update-site:1.1.0:zip failed (org.eclipse.equinox.p2.core.ProvisionException: No repository found at mvn:eu.scasefp7.eclipse:s-case-core-update-site:1.1.0:zip.)
    at org.eclipse.tycho.repository.p2base.artifact.repository.RepositoryArtifactProvider.loadRepository (RepositoryArtifactProvider.java:118)
    at org.eclipse.tycho.repository.p2base.artifact.repository.LazyArtifactRepository.getDelegate (LazyArtifactRepository.java:50)
    at org.eclipse.tycho.repository.p2base.artifact.repository.LazyArtifactRepository.contains (LazyArtifactRepository.java:87)
    at org.eclipse.tycho.repository.p2base.artifact.repository.RepositoryArtifactProvider.contains (RepositoryArtifactProvider.java:137)
    at org.eclipse.tycho.repository.local.MirroringArtifactProvider.contains (MirroringArtifactProvider.java:110)
    at org.eclipse.tycho.repository.p2base.artifact.provider.CompositeArtifactProvider.contains (CompositeArtifactProvider.java:59)
    at org.eclipse.tycho.repository.p2base.artifact.provider.CompositeArtifactProvider.getArtifactFile (CompositeArtifactProvider.java:86)
    at org.eclipse.tycho.p2.target.TargetPlatformBaseImpl.getLocalArtifactFile (TargetPlatformBaseImpl.java:130)
    at org.eclipse.tycho.p2.resolver.DefaultP2ResolutionResult.lambda$0 (DefaultP2ResolutionResult.java:58)
    at org.eclipse.tycho.p2.resolver.DefaultP2ResolutionResultEntry.getLocation (DefaultP2ResolutionResultEntry.java:73)
    at org.eclipse.tycho.p2.resolver.P2DependencyResolver.lambda$newDefaultTargetPlatform$2 (P2DependencyResolver.java:450)
    at org.eclipse.tycho.core.osgitools.targetplatform.ArtifactCollection.lambda$addArtifactFile$2 (ArtifactCollection.java:66)
    at org.eclipse.tycho.core.osgitools.DefaultArtifactDescriptor.getLocation (DefaultArtifactDescriptor.java:62)
    at org.eclipse.tycho.core.osgitools.targetplatform.ArtifactCollection.lambda$addArtifact$3 (ArtifactCollection.java:121)
    at org.eclipse.tycho.core.osgitools.DefaultArtifactDescriptor.getLocation (DefaultArtifactDescriptor.java:62)
    at org.eclipse.tycho.core.osgitools.targetplatform.ArtifactCollection.lambda$addArtifact$3 (ArtifactCollection.java:121)
    at org.eclipse.tycho.core.osgitools.DefaultArtifactDescriptor.getLocation (DefaultArtifactDescriptor.java:62)
    at org.eclipse.tycho.p2.facade.RepositoryReferenceTool.lambda$addTargetPlatformRepository$0 (RepositoryReferenceTool.java:130)
    at java.util.ArrayList.forEach (ArrayList.java:1540)
    at org.eclipse.tycho.p2.facade.RepositoryReferenceTool.addTargetPlatformRepository (RepositoryReferenceTool.java:130)
    at org.eclipse.tycho.p2.facade.RepositoryReferenceTool.getVisibleRepositories (RepositoryReferenceTool.java:98)
    at org.eclipse.tycho.plugins.p2.repository.AssembleRepositoryMojo.getVisibleRepositories (AssembleRepositoryMojo.java:193)
    at org.eclipse.tycho.plugins.p2.repository.AssembleRepositoryMojo.execute (AssembleRepositoryMojo.java:155)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.eclipse.equinox.p2.core.ProvisionException: No repository found at mvn:eu.scasefp7.eclipse:s-case-core-update-site:1.1.0:zip.
    at org.eclipse.equinox.internal.p2.repository.helpers.AbstractRepositoryManager.fail (AbstractRepositoryManager.java:397)
    at org.eclipse.equinox.internal.p2.repository.helpers.AbstractRepositoryManager.loadRepository (AbstractRepositoryManager.java:701)
    at org.eclipse.equinox.internal.p2.artifact.repository.ArtifactRepositoryManager.loadRepository (ArtifactRepositoryManager.java:131)
    at org.eclipse.equinox.internal.p2.artifact.repository.ArtifactRepositoryManager.loadRepository (ArtifactRepositoryManager.java:125)
    at org.eclipse.tycho.p2.remote.RemoteArtifactRepositoryManager.loadRepository (RemoteArtifactRepositoryManager.java:105)
    at org.eclipse.tycho.repository.p2base.artifact.repository.RepositoryArtifactProvider.loadRepository (RepositoryArtifactProvider.java:116)
    at org.eclipse.tycho.repository.p2base.artifact.repository.LazyArtifactRepository.getDelegate (LazyArtifactRepository.java:50)
    at org.eclipse.tycho.repository.p2base.artifact.repository.LazyArtifactRepository.contains (LazyArtifactRepository.java:87)
    at org.eclipse.tycho.repository.p2base.artifact.repository.RepositoryArtifactProvider.contains (RepositoryArtifactProvider.java:137)
    at org.eclipse.tycho.repository.local.MirroringArtifactProvider.contains (MirroringArtifactProvider.java:110)
    at org.eclipse.tycho.repository.p2base.artifact.provider.CompositeArtifactProvider.contains (CompositeArtifactProvider.java:59)
    at org.eclipse.tycho.repository.p2base.artifact.provider.CompositeArtifactProvider.getArtifactFile (CompositeArtifactProvider.java:86)
    at org.eclipse.tycho.p2.target.TargetPlatformBaseImpl.getLocalArtifactFile (TargetPlatformBaseImpl.java:130)
    at org.eclipse.tycho.p2.resolver.DefaultP2ResolutionResult.lambda$0 (DefaultP2ResolutionResult.java:58)
    at org.eclipse.tycho.p2.resolver.DefaultP2ResolutionResultEntry.getLocation (DefaultP2ResolutionResultEntry.java:73)
    at org.eclipse.tycho.p2.resolver.P2DependencyResolver.lambda$newDefaultTargetPlatform$2 (P2DependencyResolver.java:450)
    at org.eclipse.tycho.core.osgitools.targetplatform.ArtifactCollection.lambda$addArtifactFile$2 (ArtifactCollection.java:66)
    at org.eclipse.tycho.core.osgitools.DefaultArtifactDescriptor.getLocation (DefaultArtifactDescriptor.java:62)
    at org.eclipse.tycho.core.osgitools.targetplatform.ArtifactCollection.lambda$addArtifact$3 (ArtifactCollection.java:121)
    at org.eclipse.tycho.core.osgitools.DefaultArtifactDescriptor.getLocation (DefaultArtifactDescriptor.java:62)
    at org.eclipse.tycho.core.osgitools.targetplatform.ArtifactCollection.lambda$addArtifact$3 (ArtifactCollection.java:121)
    at org.eclipse.tycho.core.osgitools.DefaultArtifactDescriptor.getLocation (DefaultArtifactDescriptor.java:62)
    at org.eclipse.tycho.p2.facade.RepositoryReferenceTool.lambda$addTargetPlatformRepository$0 (RepositoryReferenceTool.java:130)
    at java.util.ArrayList.forEach (ArrayList.java:1540)
    at org.eclipse.tycho.p2.facade.RepositoryReferenceTool.addTargetPlatformRepository (RepositoryReferenceTool.java:130)
    at org.eclipse.tycho.p2.facade.RepositoryReferenceTool.getVisibleRepositories (RepositoryReferenceTool.java:98)
    at org.eclipse.tycho.plugins.p2.repository.AssembleRepositoryMojo.getVisibleRepositories (AssembleRepositoryMojo.java:193)
    at org.eclipse.tycho.plugins.p2.repository.AssembleRepositoryMojo.execute (AssembleRepositoryMojo.java:155)
```

--- a/tycho-bundles/org.eclipse.tycho.core.shared/src/main/java/org/eclipse/tycho/core/shared/MavenDependenciesResolver.java
+++ b/tycho-bundles/org.eclipse.tycho.core.shared/src/main/java/org/eclipse/tycho/core/shared/MavenDependenciesResolver.java
@@ -14,7 +14,6 @@ package org.eclipse.tycho.core.shared;
 
 import java.io.File;
 import java.util.Collection;
-import java.util.Collections;
 
 public interface MavenDependenciesResolver {
 
@@ -34,16 +33,21 @@ public interface MavenDependenciesResolver {
      * @param dependencyScope
      *            optional dependency scope, if given it tries to resolve transitive dependencies of
      *            the given artifact as well
+     * @param additionalRepositories
+     *            additional repositories to use in the resolve process
      * @return
      */
     default Collection<? /* IArtifactFacade */> resolve(String groupId, String artifactId, String version,
-            String packaging, String classifier, String dependencyScope) {
-        return resolve(groupId, artifactId, version, packaging, classifier, dependencyScope, Collections.emptyList());
+            String packaging, String classifier, String dependencyScope,
+            Collection<MavenArtifactRepositoryReference> additionalRepositories) {
+        return resolve(groupId, artifactId, version, packaging, classifier, dependencyScope, additionalRepositories,
+                null);
     }
 
     Collection<? /* IArtifactFacade */> resolve(String groupId, String artifactId, String version, String packaging,
             String classifier, String dependencyScope,
-            Collection<MavenArtifactRepositoryReference> additionalRepositories);
+            Collection<MavenArtifactRepositoryReference> additionalRepositories,
+            Object/* MavenSession */ session);
 
     File getRepositoryRoot();
 

--- a/tycho-bundles/org.eclipse.tycho.p2.resolver.impl/src/main/java/org/eclipse/tycho/p2/resolver/MavenTargetDefinitionContent.java
+++ b/tycho-bundles/org.eclipse.tycho.p2.resolver.impl/src/main/java/org/eclipse/tycho/p2/resolver/MavenTargetDefinitionContent.java
@@ -17,6 +17,7 @@ import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.Enumeration;
 import java.util.HashMap;
 import java.util.Iterator;
@@ -181,7 +182,7 @@ public class MavenTargetDefinitionContent implements TargetDefinitionContent {
                             || (sourceMode == IncludeSourceMode.honor && location.includeSource())) {
                         Collection<?> sourceArtifacts = mavenDependenciesResolver.resolve(mavenArtifact.getGroupId(),
                                 mavenArtifact.getArtifactId(), mavenArtifact.getVersion(),
-                                mavenArtifact.getPackagingType(), "sources", null);
+                                mavenArtifact.getPackagingType(), "sources", null, Collections.emptyList());
                         Iterator<IArtifactFacade> sources = sourceArtifacts.stream()
                                 .filter(IArtifactFacade.class::isInstance).map(IArtifactFacade.class::cast).iterator();
                         while (sources.hasNext()) {

--- a/tycho-core/src/main/java/org/eclipse/tycho/osgi/configuration/MavenDependenciesResolverConfigurer.java
+++ b/tycho-core/src/main/java/org/eclipse/tycho/osgi/configuration/MavenDependenciesResolverConfigurer.java
@@ -16,6 +16,7 @@ import java.io.File;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
+import java.util.Objects;
 import java.util.Set;
 
 import org.apache.maven.artifact.Artifact;
@@ -49,7 +50,8 @@ public class MavenDependenciesResolverConfigurer extends EquinoxLifecycleListene
 
     @Override
     public Collection<?> resolve(String groupId, String artifactId, String version, String packaging, String classifier,
-            String dependencyScope, Collection<MavenArtifactRepositoryReference> additionalRepositories) {
+            String dependencyScope, Collection<MavenArtifactRepositoryReference> additionalRepositories,
+            Object session) {
         Artifact artifact;
         if (classifier != null && !classifier.isEmpty()) {
             artifact = repositorySystem.createArtifactWithClassifier(groupId, artifactId, version, packaging,
@@ -60,17 +62,21 @@ public class MavenDependenciesResolverConfigurer extends EquinoxLifecycleListene
         }
         ArtifactResolutionRequest request = new ArtifactResolutionRequest();
         request.setArtifact(artifact);
-        MavenSession session = context.getSession();
-        request.setOffline(session.isOffline());
-        request.setLocalRepository(session.getLocalRepository());
+        MavenSession mavenSession = getMavenSession(session);
+        request.setOffline(mavenSession.isOffline());
+        request.setLocalRepository(mavenSession.getLocalRepository());
         request.setResolveTransitively(dependencyScope != null && !dependencyScope.isEmpty());
-        List<ArtifactRepository> repositories = new ArrayList<>(
-                session.getCurrentProject().getRemoteArtifactRepositories());
-        for (MavenArtifactRepositoryReference reference : additionalRepositories) {
-            repositories.add(
-                    repositorySystem.createArtifactRepository(reference.getId(), reference.getUrl(), null, null, null));
+        if (additionalRepositories != null && additionalRepositories.size() > 0) {
+            List<ArtifactRepository> repositories = new ArrayList<>(
+                    mavenSession.getCurrentProject().getRemoteArtifactRepositories());
+            for (MavenArtifactRepositoryReference reference : additionalRepositories) {
+                repositories.add(repositorySystem.createArtifactRepository(reference.getId(), reference.getUrl(), null,
+                        null, null));
+            }
+            request.setRemoteRepositories(repositorySystem.getEffectiveRepositories(repositories));
+        } else {
+            request.setRemoteRepositories(mavenSession.getCurrentProject().getRemoteArtifactRepositories());
         }
-        request.setRemoteRepositories(repositories);
         ArtifactResolutionResult result = repositorySystem.resolve(request);
         Set<Artifact> artifacts = result.getArtifacts();
         ArrayList<IArtifactFacade> list = new ArrayList<IArtifactFacade>();
@@ -80,6 +86,17 @@ public class MavenDependenciesResolverConfigurer extends EquinoxLifecycleListene
         return list;
     }
 
+    protected MavenSession getMavenSession(Object session) {
+        MavenSession mavenSession;
+        if (session instanceof MavenSession) {
+            mavenSession = (MavenSession) session;
+        } else {
+            mavenSession = Objects.requireNonNull(context.getSession(),
+                    "Can't acquire maven session from context, called outside maven thread context?");
+        }
+        return mavenSession;
+    }
+
     @Override
     public void afterFrameworkStarted(EmbeddedEquinox framework) {
         framework.registerService(MavenDependenciesResolver.class, this);
@@ -87,8 +104,7 @@ public class MavenDependenciesResolverConfigurer extends EquinoxLifecycleListene
 
     @Override
     public File getRepositoryRoot() {
-
-        return new File(context.getSession().getLocalRepository().getBasedir());
+        return new File(getMavenSession(null).getLocalRepository().getBasedir());
     }
 
 }

--- a/tycho-core/src/main/java/org/eclipse/tycho/osgi/configuration/MavenProtocolHandler.java
+++ b/tycho-core/src/main/java/org/eclipse/tycho/osgi/configuration/MavenProtocolHandler.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2020 Christoph Läubrich and others.
+ * Copyright (c) 2020, 2021 Christoph Läubrich and others.
  * This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License 2.0
  * which accompanies this distribution, and is available at
@@ -20,9 +20,13 @@ import java.net.URL;
 import java.net.URLConnection;
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.Hashtable;
 
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.LegacySupport;
 import org.codehaus.plexus.component.annotations.Component;
+import org.codehaus.plexus.component.annotations.Requirement;
 import org.eclipse.sisu.equinox.EquinoxServiceFactory;
 import org.eclipse.sisu.equinox.embedder.EmbeddedEquinox;
 import org.eclipse.sisu.equinox.embedder.EquinoxLifecycleListener;
@@ -35,25 +39,32 @@ import org.osgi.service.url.URLStreamHandlerService;
 @Component(role = EquinoxLifecycleListener.class, hint = "MavenProtocolHandler")
 public class MavenProtocolHandler extends EquinoxLifecycleListener {
 
+    @Requirement
+    private LegacySupport context;
+
     @Override
     public void afterFrameworkStarted(EmbeddedEquinox framework) {
 
         Hashtable<String, Object> properties = new Hashtable<>();
         properties.put(URLConstants.URL_HANDLER_PROTOCOL, new String[] { "mvn" });
         framework.registerService(URLStreamHandlerService.class,
-                new MvnProtocolHandlerService(framework.getServiceFactory()), properties);
+                new MvnProtocolHandlerService(framework.getServiceFactory(), context.getSession()), properties);
+
     }
 
     private class MvnProtocolHandlerService extends AbstractURLStreamHandlerService {
 
         private EquinoxServiceFactory serviceFactory;
+        private MavenSession mavenSession;
 
-        public MvnProtocolHandlerService(EquinoxServiceFactory serviceFactory) {
+        public MvnProtocolHandlerService(EquinoxServiceFactory serviceFactory, MavenSession mavenSession) {
             this.serviceFactory = serviceFactory;
+            this.mavenSession = mavenSession;
         }
 
         public URLConnection openConnection(URL url) {
-            return new MavenURLConnection(url, serviceFactory.getService(MavenDependenciesResolver.class));
+            return new MavenURLConnection(url, serviceFactory.getService(MavenDependenciesResolver.class),
+                    mavenSession);
         }
     }
 
@@ -62,47 +73,55 @@ public class MavenProtocolHandler extends EquinoxLifecycleListener {
         private MavenDependenciesResolver resolver;
         private String subPath;
         private IArtifactFacade artifactFacade;
+        private MavenSession session;
 
-        protected MavenURLConnection(URL url, MavenDependenciesResolver dependenciesResolver) {
+        protected MavenURLConnection(URL url, MavenDependenciesResolver dependenciesResolver,
+                MavenSession mavenSession) {
             super(url);
             this.resolver = dependenciesResolver;
+            this.session = mavenSession;
         }
 
         @Override
         public void connect() throws IOException {
-            if (artifactFacade != null) {
-                return;
+            try {
+                if (artifactFacade != null) {
+                    return;
+                }
+                if (resolver == null) {
+                    throw new IOException("resolver service is not available");
+                }
+                String path = url.getPath();
+                if (path == null) {
+                    throw new IOException("maven coordinates are missing");
+                }
+                int subPathIndex = path.indexOf('/');
+                String[] coordinates;
+                if (subPathIndex > -1) {
+                    subPath = path.substring(subPathIndex);
+                    coordinates = path.substring(0, subPathIndex).split(":");
+                } else {
+                    coordinates = path.split(":");
+                }
+                if (coordinates.length < 3) {
+                    throw new IOException("required format is groupId:artifactId:version[:packaging[:classifier]]");
+                }
+                String type = coordinates.length > 3 ? coordinates[3] : "jar";
+                String classifier = coordinates.length > 4 ? coordinates[4] : null;
+                Collection<?> resolve = resolver.resolve(coordinates[0], coordinates[1], coordinates[2], type,
+                        classifier, null, Collections.emptyList(), session);
+                if (resolve.isEmpty()) {
+                    throw new IOException("artifact " + Arrays.toString(coordinates)
+                            + " could not be downloaded from any of the available repositories");
+                }
+                if (resolve.size() > 1) {
+                    throw new IOException(
+                            "artifact " + Arrays.toString(coordinates) + " resolves to multiple artifacts");
+                }
+                artifactFacade = (IArtifactFacade) resolve.iterator().next();
+            } catch (RuntimeException e) {
+                throw new IOException("internal error connecting to maven url " + url, e);
             }
-            if (resolver == null) {
-                throw new IOException("resolver service is not available");
-            }
-            String path = url.getPath();
-            if (path == null) {
-                throw new IOException("maven coordinates are missing");
-            }
-            int subPathIndex = path.indexOf('/');
-            String[] coordinates;
-            if (subPathIndex > -1) {
-                subPath = path.substring(subPathIndex);
-                coordinates = path.substring(0, subPathIndex).split(":");
-            } else {
-                coordinates = path.split(":");
-            }
-            if (coordinates.length < 3) {
-                throw new IOException("required format is groupId:artifactId:version[:packaging[:classifier]]");
-            }
-            String type = coordinates.length > 3 ? coordinates[3] : "jar";
-            String classifier = coordinates.length > 4 ? coordinates[4] : null;
-            Collection<?> resolve = resolver.resolve(coordinates[0], coordinates[1], coordinates[2], type, classifier,
-                    null);
-            if (resolve.isEmpty()) {
-                throw new IOException("artifact " + Arrays.toString(coordinates)
-                        + " could not be downloaded from any of the available repositories");
-            }
-            if (resolve.size() > 1) {
-                throw new IOException("artifact " + Arrays.toString(coordinates) + " resolves to multiple artifacts");
-            }
-            artifactFacade = (IArtifactFacade) resolve.iterator().next();
         }
 
         @Override

